### ISSUE TITLE
Layout Grid Block: word wrap breaks words on smaller screens if the font-size is too big 

### ISSUE THREAD

**Issue by MaggieCabrera** [Automattic/block-experiments#155]
#### Steps to reproduce the issue

1. Create a Layout Grid Block
2. Add an H1 heading, use the biggest word you can think off :) 
3. Save and check on very small resolution such of an iphone 5 or similar.

Related issue: https://github.com/Automattic/themes/issues/2666

#### What I expected

I expect to see the whole word on one line.

#### What happened instead

**Screenshots**

![](https://d.pr/i/R8OruW/RGFCYOeHMv+)

![](https://user-images.githubusercontent.com/3593343/98096681-5679a700-1e8c-11eb-87e3-66580a41ee86.png)

**Comment by jasmussen** [comment]
Great ticket, thank you!

So this is an interesting one, it's kind of a "choose your poison". Specifically, this section of code from Layout Grid appears responsible:

```
.wp-block-jetpack-layout-grid .wp-block-jetpack-layout-grid-column * {
    word-break: break-word;
    word-wrap: break-word;
}
```

What it does, is this:

<img width="1270" alt="Screenshot 2020-11-16 at 09 27 00" src="https://user-images.githubusercontent.com/1204802/99229223-07b60080-27ee-11eb-8eb9-545287e21a1a.png">

You can see a heading with the same silly headline right below the colored boxes, showsing that by default, headings don't break words. 

However here's what happens if we remove that rule:

<img width="1270" alt="Screenshot 2020-11-16 at 09 26 48" src="https://user-images.githubusercontent.com/1204802/99229298-1c929400-27ee-11eb-968a-6dc89e5e0b09.png">

In my testing just now, it appears to at least overflow like that, consistently across Firefox, Safari, Chrome. Additionally, it doesn't _appear_ to break the layout, though I could swear it did at one point and that was the cuse of the addition. There's more context in #34, cc: @johngodley 

The rule applies to headings and body text, but the layout doesn't appear to expand when removing it:

<img width="1270" alt="Screenshot 2020-11-16 at 09 33 34" src="https://user-images.githubusercontent.com/1204802/99229891-e9043980-27ee-11eb-8dd9-9a81444c2e21.png">

Bottom line — I'm happy to remove or tweak this rule, becuase it _seems_ like we can, but are we sure that's what we want?

**Comment by MaggieCabrera** [comment]
I'm not really sure about this one, because with/without the rule we are going to have an issue. I'm not sure if @ianstewart you were following this?

**Comment by glendaviesnz** [comment]
@ianstewart - did you have any views on wrapping or hiding for this. 

The issue i see with with changing it to overflow instead is the potentially of breaking existing content where customers may be happy with the word break and their site users can see the full content, and then suddenly part of the content will be cut off.

**Comment by jasmussen** [comment]
Outside of doing nothing at all, I only really see two alternatives:

- Find better word-wrap CSS. I think we're already using best practices though.
- Make the word wrap a toggle in the layout grid block. I don't love this, as it only lets you choose between the lesser of two evils.

It seems the examle shown is just an unfortunate responsive case of too large text on too small a screen, and that the real solution is for patterns to come with better built-in responsiveness. That is, we need something like https://github.com/WordPress/gutenberg/issues/19909 to address this at the source.

**Comment by glendaviesnz** [comment]
I agree that there isn't really a good solution to this, and hard to judge which is the lesser of the two evils of the possible alternatives. 

My suggestion is that the real solution is as @jasmussen mentions to give customers a way to easily see how things look in different viewports so they can make adjustments to font sizes, etc. if they wish.

In the mean time I suggest that the only change we make is to add a hyphen to word breaks in those browsers that support `hyphens: auto;` so at least it is a clearer that a word is broken across lines, eg:

<img width="416" alt="Screen Shot 2020-12-10 at 1 54 09 PM" src="https://user-images.githubusercontent.com/3629020/101707278-ca444a80-3aef-11eb-81a5-7829aa439151.png">

I think this is better than unexpectedly overflowing and hiding the text. Any other thoughts?

**Comment by glendaviesnz** [comment]
Have added a PR for the hyphen at https://github.com/Automattic/block-experiments/pull/159

**Comment by jasmussen** [comment]
Love it. Just encountered the issue here 🙈 

![Screenshot_20201210-084010](https://user-images.githubusercontent.com/1204802/101740855-a79f3b00-3ac9-11eb-8fbb-4de206ec4127.png)

To me this reaffirms that the solution is to be able to provide a smaller font size for the smaller breakpoints. For which we still need a variant of https://github.com/WordPress/gutenberg/issues/19909. 

But the hyphen is a great interim solution. Thank you Glen!

**Comment by apeatling** [comment]
Closing this one out since the typo fix is deployed. :+1:

**Comment by jasmussen** [comment]
Per conversation in https://github.com/Automattic/block-experiments/issues/173, I've had to create a PR to remove hyphenation again.
